public/app/types/unified-alerting.ts

```diff
@@ -158,7 +158,7 @@ interface EvalMatch {
 }
 
 export interface StateHistoryItemData {
-  noData: boolean;
+  noData?: boolean;
```

Comment:
why this noData is not required anymore?

Reply:
It seems like I introduced this initially in https://github.com/grafana/grafana/pull/42362/files#diff-0d52198fda745d8666c1d37454ef6a8b423b8e16843962d944bb0d0eac9f997aR151 but re-examining this now there's no reason why it shouldn't be optional as the API does return empty data objects (`{}`)